### ISSUE TITLE
Incorrect guard checking when from is not provided

### ISSUE THREAD

**Issue by rafaelsales** [aasm/aasm#454]
Version: 4.12.0

Expected this to work
```
event :work do
  transitions to: :foo, guard: :should_foo?
  transitions to: :bar, guard: :should_bar?
end
```
When `should_foo` returns `false`, the `should_bar?` is not even called, but when `from` is provided, the `should_bar?` is checked.
```
event :work do
  transitions from: :initial, to: :foo, guard: :should_foo?
  transitions from: :initial, to: :bar, guard: :should_bar?
end
```

**Comment by adam-szczombrowski** [comment]
Version: 4.12.0

I've encountered the same problem. IMO the behaviour should be the same whether you specify the `:from` option or not. Will try to come up with a pr solving this issue.

**Comment by adam-szczombrowski** [comment]
@anilmaurya can you confirm that this is a bug? Maybe it's desired behaviour and I don't understand something.

**Comment by anilmaurya** [comment]
@adam-szczombrowski I agree that behaviour should be  same whether you specify the `:from `option or not . Looking forward to your PR.

Thank you for helping 😊

**Comment by anilmaurya** [comment]
Fixed in #456 , thanks to @adam-szczombrowski
